### DISCUSSION TITLE
Free-sort Plugin Modification

### DISCUSSION THREAD

**Discussion by Shelka4444** [jspsych/jsPsych#2656] [Q&A]
Hi there, I'm on the newer end of coding, and I am looking for advice on how to modify the free-sort plugin. So, what we want to do now is a free-sort task where the participant is asked to drop agents into clusters based on similarity. Does anyone have any tips for how to be able to a) assess if there are any clusters, and then b) if a cluster is present, grab all the agents in that cluster and run another sort for each available grouping. I hope that makes sense. I'm not looking for the actual code (though that is helpful), but rather how to approach this type of problem. Thank you!

**Comment by sophief10** [comment]
Hi! I am also a new coder, so I probably can't help much 😅. However, based on your question, I think you might find these two resources useful, if you haven't already seen them:

- [article on plugin development in jsPsych](https://www.jspsych.org/7.2/developers/plugin-development/) 
- [tutorial on creating draggable HTML elements](https://www.w3schools.com/howto/howto_js_draggable.asp)

In case you don't know, I'll tell you that anything you can do with JavaScript (and HTML and CSS), you can also do with jsPsych. If you are new to coding in general, you may find it helpful to draw a flowchart of how you want users to progress through your experiment. At each step in the flowchart, note what the user should see on their screen, how they should interact with or respond to what they see, and what kind of data you want to collect from them. Then, you can come up with some code to achieve what you want. HTML and CSS code can generally control what the user sees on their screen, and choosing the appropriate jsPsych plugins will allow you to specify how the user should interact with the screen and generate data. If you want some more sophisticated functionality, you could play around with [timelines](https://www.jspsych.org/7.2/overview/timeline/), [reference methods](https://www.jspsych.org/7.2/reference/jspsych/), etc.
